RV-Monitor/Kernel_Safety_Claims_by_Runtime_Verification_Monitors.md

```diff
@@ -0,0 +1,542 @@
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >heartbeat_check
+   </td>
+   <td rowspan="3" >Before ioctl returns success, the RV monitor checks if the operations on the HW have been performed as expected 
+   </td>
+   <td>success: the operations on the HW have been performed as expected
+   </td>
+   <td>beating
+   </td>
+  </tr>
+  <tr>
+   <td>failure: ioctl was going to return success while one or more operations on the HW were not as expected   
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >beating
+   </td>
+   <td rowspan="2" >The ioctl returned success and the i6300 watchdog is ticking according to the programmed timeout
+   </td>
+   <td>write: the Safety App invoked a write syscall on the i6300 file descriptor
+   </td>
+   <td>reload
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >reload
+   </td>
+   <td rowspan="3" >The watchdog is reloaded according to the current timeout and starts ticking
+   </td>
+   <td>failure: something went wrong when reloading the watchdog
+   </td>
+   <td>beating
+   </td>
+  </tr>
+  <tr>
+   <td>success: the reload operation succeeds and write is going to return success
+   </td>
+   <td>reload_check
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >reload_check
+   </td>
+   <td rowspan="3" >Before write returns success, the RV monitor checks if the operations on the HW have been performed as expected 
+   </td>
+   <td>success: the operations on the HW have been performed as expected
+   </td>
+   <td>beating
+   </td>
+  </tr>
+  <tr>
+   <td>failure: write was going to return success while one or more operations on the HW were not as expected   
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td>BPA / unlock_wr: the watchdog timer registers were attempted to be written or a thread different from the safety app tried to unlock the i6300 HW for write access
+   </td>
+   <td>Fail
+   </td>
+  </tr>
+  <tr>
+   <td>fail
+   </td>
+   <td>The monitored code is in an invalid state, hence the RV Monitor triggers Kernel Panic here 
+   </td>
+   <td>None
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+
+# Limitations and Improvements
+
+
+## Tracepoints
+
+**TODO**: evaluate if it is safer OR easier (in terms of FuSa effort) to resolve monitoring points at compile time instead of using runtime tracepoints.
```

Comment:
compile-time can be verified statically - hard to beat that :-D

Reply:
noted....but is it OK to leave the TODO as written above or you want to propose a modification in the text ?

Reply:
I would prefer to say it explicitly that compile time is the preferred approach, because the other option is not equally safe (and good luck proving that it is *sufficiently* safe)

Reply:
Ok I think we can remove the TODO and just state:
<<Relying on runtime tracepoints implies allocating safety requirements to a mechanisms that is more complex than relying on compile time resolution of monitoring points hooks; therefore it may be preferable to use compile time ones. However each user can make his own analysis in this regard>>. Ok ?

Reply:
See https://github.com/elisa-tech/Safety_Architecture_WG/pull/13#discussion_r1494805867